### ISSUE TITLE
DISCUSSION: Splitting BCS into core and scripts

### ISSUE THREAD

**Issue by thisIsMikeKane** [ecobee/building-controls-simulator#134]
As BCS moves from alpha to beta, its user base will be less interested in development and more interested in using BCS. Should we consider spitting BCS into the *bcs-core* of the docker container and everything needed to run the *bcs-simulator* . For example, if the `data` and `scripts` directories made up *bcs-simulator* were in a separate repo that was a submodule to *bcs-core* then it would be more clear to day-to-day users what they should be 'allowed' to touch. This would also more clearly enable branches/forks of scripts without needing to have all of the *bcs-core* in the repo. 

tldr; Moving the day-to-day scripts of BCS to their own repo would reduce the learning curve for new users.

**Comment by tomstesco** [comment]
Yupe that's the long term plan, ideally we can package the BCS as a python library at some point. This is complicated by the dependencies on difficult to configure libraries such as Energyplus, PyFMI, and jupyter-lab/plotly.

For the mid-term I think a good middle ground is using the Docker images i'll be releasing on Dockerhub at https://hub.docker.com/r/tstesco/building-controls-simulator/tags. Then we can have application repos that mount into the Docker container containing examples and common use-cases built on top of the BCS. This is how we're using the BCS internally at ecobee. This will be somewhat similar to the notebook examples already in the BCS repo.

**Comment by thisIsMikeKane** [comment]
One thing that is one my mind/vision is parallelization and service-ization of BCS. 

Instead of a scripting repo mounting inside of BCS, what about having the scripting repo as the supervisor. If the container is deployed without mounting, then the script could deploy many containers to have simulations run in parallel. I could envision a scripting wrapper for BCS that either runs it in **interactive** mode, mounting the scripting repo as we do now; a **batch** mode that may run multiple BCS containers in parallel to run a batch of multiple simulations; and a **service** mode that could manage a swarm of BCS containers as a load balancer for simulations as a service. 

As a next step towards this vision, do you @tomstesco see any challenge with removing the `volumes` mounting in `docker-compose.yml` ?

**Comment by tomstesco** [comment]
No challenges with the volume mounting, this just gives us flexibility in the interactive mode we use today - which will become our client implementation. The batch and service mode would likely be the same implementation, just depending on if servers were persistent or ephemeral - this would be the server implementation. I've done some preliminary work with dask (https://dask.org/) and will likely use something similar to allow for a client running locally to create a `simulation_server` object that contains all the instructions to get the data and run the simulations (lazily initialized), then send that object to the remote server implementation to schedule across compute nodes and actually run. The resulting data will be stored as configured in the `DataDestination` object.
